src/notifications.c

```diff
@@ -384,10 +392,15 @@ void ClusterSlotMigrationEvent(RedisModuleCtx *ctx, RedisModuleEvent eid, uint64
   }
 }
 
+static void drainHighPriorityJobs(void) {
+  workersThreadPool_DrainHighPriority(RSDummyContext);
+}

```

Comment:
Why pass it as a pointer, and not make it part of `ASM_StateMachine_StartTrim`? If there is any justification for the new H file, it’s the call to this function

Reply:
the new H file is created to allow unit testing. The passing as pointer is to allow testing without the dependency of needing all the workers logic